src/acp/rule.test.ts

```diff
@@ -702,3 +783,778 @@ describe("getRequiredRulesOnPolicyAll", () => {
+    const rule = mockRule("https://my.pod/rule-resource#rule", {
+      agents: ["https://my.pod/profile#me", "https://your.pod/profile#you"],
+      public: true,
+      authenticated: true,
+    });
+    const result = removeAgentFromRule(rule, "https://your.pod/profile#you");
+    expect(
+      result.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://my.pod/rule-resource#rule"),
+          DataFactory.namedNode(ACP_AGENT),
+          DataFactory.namedNode("https://your.pod/profile#you")
+        )
+      )
+    ).toBe(false);
+    expect(
+      result.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://my.pod/rule-resource#rule"),
+          DataFactory.namedNode(ACP_AGENT),
+          DataFactory.namedNode("https://my.pod/profile#me")
+        )
+      )
+    ).toBe(true);
+    expect(
+      result.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://my.pod/rule-resource#rule"),
+          DataFactory.namedNode(ACP_AGENT),
+          DataFactory.namedNode(ACP_PUBLIC)
+        )
+      )
+    ).toBe(true);
+    expect(
+      result.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://my.pod/rule-resource#rule"),
+          DataFactory.namedNode(ACP_AGENT),
+          DataFactory.namedNode(ACP_AUTHENTICATED)
+        )
+      )
+    ).toBe(true);
+  });
+
+  it("does not remove groups, even with matching IRI", () => {
+    const rule = mockRule("https://my.pod/rule-resource#rule", {
+      groups: ["https://my.pod/groups#a-group"],
+    });
+    const result = removeAgentFromRule(rule, "https://my.pod/groups#a-group");
+    expect(
+      result.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://my.pod/rule-resource#rule"),
+          DataFactory.namedNode(ACP_GROUP),
+          DataFactory.namedNode("https://my.pod/groups#a-group")
+        )
+      )
+    ).toBe(true);
+  });
+});
+
+describe("getGroupForRuleAll", () => {
+  it("returns all the groups a rule applies to", () => {
+    const rule = mockRule("https://my.pod/rule-resource#some-rule", {
+      groups: [
+        "https://my.pod/groups#a-group",
+        "https://my.pod/groups#another-group",
+      ],
+    });
+    const groups = getGroupForRuleAll(rule);
+    expect(groups).toContainEqual("https://my.pod/groups#a-group");
+    expect(groups).toContainEqual("https://my.pod/groups#another-group");
+  });
+
+  it("does not return the agents/public/authenticated a rule applies to", () => {
+    const rule = mockRule("https://my.pod/rule-resource#some-rule", {
+      agents: ["https://my.pod/profile#me"],
+      public: true,
+      authenticated: true,
+    });
+    const groups = getGroupForRuleAll(rule);
+    expect(groups).not.toContainEqual("https://my.pod/profile#me");
+    expect(groups).not.toContainEqual(ACP_AUTHENTICATED);
+    expect(groups).not.toContainEqual(ACP_PUBLIC);
+  });
+});
+
+describe("setGroupInRule", () => {
```

Comment:
Just wondering: would it be a good idea to have consistency of either `getGroupForRuleAll()`/`setGroupForRule()` or `getGroupInRuleAll()`/`setGroupInRule()`? Or do the different prepositions just make more sense in the different context? (Not being a native speaker getting me down here...)

Reply:
@pmcb55 , could you have a look at the function names in the API ? I initially thought the prepositions would make sense in their context, but I'd prefer have that double-checked :)

Reply:
Yeah, Vinnie's right - `getGroupForRuleAll()` / `setGroupForRule()` are more appropriately named.

Reply:
Fixed in 9107fe466f0f0d056bdb8146e6e14f59347f0213